### DISCUSSION TITLE
Replace Port 8443 with other ports

### DISCUSSION THREAD

**Discussion by git-praveen-kumar** [Kong/kubernetes-ingress-controller#6426] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Port 8443 is used

### Expected Behavior

Replace port 8443 with other ports

### Steps To Reproduce

```markdown
Kong Ingress Controller is deployed using Helm chart
```

### Kong Ingress Controller version

```shell
v3.6
```

### Kubernetes version

```shell
v1.28.5
```

### Anything else?

Kong Ingress controller is deployed using Helm chart, Since Port 8443 is being used by other application. I need to replace that Port  with other Ports.

**Comment by pmalek** [comment]
Please share the values.yaml that you've used to deploy. Have you used [`kong` chart](https://github.com/Kong/charts/tree/main/charts/kong)? You can configure this port using https://github.com/Kong/charts/blob/fe3c070b614dea6db0ed6517871c3dfdc75b0516/charts/kong/values.yaml#L317 `proxy.tls.containerPort`

**Comment by git-praveen-kumar** [comment]
@pmalek  Thanks for quick response, Tired the above changes already. Results the kong pods in crashloopbackoff
![Screenshot 2024-08-16 153330](https://github.com/user-attachments/assets/d85e4f2b-1898-4b5f-9efe-65094a683f2c)
![Screenshot 2024-08-16 153129](https://github.com/user-attachments/assets/80edeeb1-7245-412b-bdaa-01ab3711f671)

**Comment by pmalek** [comment]
I have just tried this with the latest version of the chart and it worked fine. We'll need a bit more information than just those screenshots to asses.

Please provide 
- the version of the helm chart that you're using (output from `helm ls -A` ideally)
- values.yaml file used for this deployment
- kong and ingress controller logs
- `kubectl describe` of failing pods

**Comment by git-praveen-kumar** [comment]
Please find the attached logs and screenshot
![Capture-2](https://github.com/user-attachments/assets/d69ecd02-357e-49b6-8d7a-17974c152434)
![Capture-1](https://github.com/user-attachments/assets/4e7da54a-1380-4f88-bfe7-99d024277c2d)
![Capture](https://github.com/user-attachments/assets/76eaa990-5193-42e2-83c4-b323cd8fb77a)
[kong-values.zip](https://github.com/user-attachments/files/16638095/kong-values.zip)

**Comment by pmalek** [comment]
To get the logs please execute:

```
kubectl logs -n <NAMESPACE> deployment/<KONG_DEPLOYMENT_NAME>
```

and attach the output. The readiness probes seems to be failing for some reason. 

---

Please note: your values.yaml looks like a copy paste from https://github.com/Kong/charts/blob/main/charts/kong/values.yaml with several fields changed. This is not how Helm values should be used. You should only specify the entries to you intend to override on top of the base values.yaml provided by Kong (https://github.com/Kong/charts/blob/main/charts/kong/values.yaml).

**Comment by git-praveen-kumar** [comment]
We are using same values.yaml by overriding
[Downloads.zip](https://github.com/user-attachments/files/16658768/Downloads.zip)
 the required values. Please find the attached values.yaml.
Also logs of proxy and controller pods are attached.

**Comment by pmalek** [comment]
Looking at the `values.yaml` I can still see that it contains almost all of the defaults from https://github.com/Kong/charts/blob/main/charts/kong/values.yaml. You should only specify the values that you're actually overriding. Do not copy paste all and then change those that you want to amend.

---


Looking at the proxy logs I can see 

```
ERROR: ld.so: object '/opt/dynatrace/oneagent-paas/agent/lib64/liboneagentproc.so' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
```

which indicates that your environment has some broken `LD_PRELOAD` configuration that prevents Kong from starting. Please fix that and then we can take a look again.
